devdoc/DevTestWithLDAP.md

```diff
@@ -0,0 +1,146 @@
+
+First we will set up a Docker container with a minimal 389ds LDAP Directory Server and a phpLdapAdmin (PLA) client to administer it. The PLA can be used to inspect your DS and perform any administrative tasks not supported by the ECL Watch client.
+
+Then we'll cover the configuration changes to deploy a minimal HPCC platform, both non-containerized (bare-metal) and under Kubernetes, that uses 389ds for security.
+
+## Initial State
+
+When the platform launches it ensures the LDAP DS is initialized with "OUs" (organizational units) needed. OUs for groups, users and esp authorization resources are created (among others). An HPCC Admin user is created and added to the administrator's group.
+
+If you need any other security settings beyond that, such as non-administrative users and groups, you'll need to create them using ECL Watch. The administrator account itself has the privileges to make these kinds of changes. You may see that the admin doesn't by default have _all_ permissions, but it can add any it needs to the admin group.
+
+This setup has been tested with ECL Watch authentication and authorization. It should also work for file and workunit scopes, but that hasn't been confirmed.
+
+Your customized setup will be saved to a persistent volume, and you can separately maintain different versions as needed.
+
+# Docker LDAP Directory Service and Admin Setup
+
+We'll run our 389ds and PLA in a Docker container launched from platform source: `dockerfiles/examples/ldap`.
+
+These instructions create a persistent volume for the DS inside your directory, but you can customize the docker-compose file to use another location. All of the server state is stored in the mounted volume, so if you want to be able to easily switch between different configurations you can easily do so by switching the mounted directory at launch.
+
+1. Edit the `dockerfiles/examples/ldap/docker-compose.yaml` file, customizing the admin password and PVC mount point:
+    - Change `DS_DM_PASSWORD: "<directory_manager_pw>"` placeholder with the real password you want to use. This is the password that you will use in PLA to administer the DS, and that the HPCC platform will use to interact with it using LDAP.
+    - Under volumes, replace `${HOME}/389ds-data` with the location of your choice.
+2. From `dockerfiles/examples/ldap`, run:
+
+    ```bash
+    docker compose up -d
+    ```
+3. Create the database backend on the DS. This should be done only once- the first time you run the container with a new directory mounted:
+
+    ```bash
+    docker exec -i -t 389ds /usr/sbin/dsconf localhost backend create --suffix dc=example,dc=com --be-name userRoot
+    ```
```

Comment:
You could add a "Check the location specified in 1 above" to make sure it initialized correctly. Correctly means, I guess, that is created a bunch of stuff there